opm/simulators/flow/BlackoilModelEbos.hpp

```diff
@@ -556,19 +557,21 @@ namespace Opm {
                     }
                     for (unsigned phaseIdx = 0; phaseIdx < FluidSystem::numPhases; ++ phaseIdx) {
                         Scalar tmpSat = saturationsNew[phaseIdx] - saturationsOld[phaseIdx];
-                        resultDelta += tmpSat*tmpSat;
-                        resultDenom += saturationsNew[phaseIdx]*saturationsNew[phaseIdx];
-                        assert(std::isfinite(resultDelta));
-                        assert(std::isfinite(resultDenom));
+                        resultDeltaSaturation += tmpSat*tmpSat;
+                        resultDenomSaturation += saturationsNew[phaseIdx]*saturationsNew[phaseIdx];
+                        assert(std::isfinite(resultDeltaSaturation));
+                        assert(std::isfinite(resultDenomSaturation));
                     }
                 }
             }
 
-            resultDelta = gridView.comm().sum(resultDelta);
-            resultDenom = gridView.comm().sum(resultDenom);
+            resultDeltaPressure = gridView.comm().sum(resultDeltaPressure);
+            resultDenomPressure = gridView.comm().sum(resultDenomPressure);
+            resultDeltaSaturation = gridView.comm().sum(resultDeltaSaturation);
+            resultDeltaSaturation = gridView.comm().sum(resultDeltaSaturation);
 
-            if (resultDenom > 0.0)
-                return resultDelta/resultDenom;
+            if (resultDenomPressure > 0.0 || resultDenomSaturation > 0.0)
```

Comment:
Did you mean `&&` here?  Using `||` we risk having one of the denominators be zero from which we'll either generate infinity or NaN and for what it's worth, the original code did not behave that way.

Reply:
Yes, I did! Thank you for spotting that error. I've fixed it now.